src/aws-cpp-sdk-core/include/aws/core/utils/stream/AwsChunkedStream.h

```diff
@@ -30,30 +30,28 @@ class AwsChunkedStream {
 
   size_t BufferedRead(char *dst, size_t amountToRead) {
     assert(dst != nullptr);
-    if (dst == nullptr) {
-      AWS_LOGSTREAM_ERROR("AwsChunkedStream", "dst is null");
-    }
 
-    // the chunk has ended and cannot be read from
-    if (m_chunkEnd) {
-      return 0;
-    }
+    // only read and write to chunked stream if the underlying stream
+    // is still in a valid state
+    if (m_stream->good()) {
+      // Try to read in a 64K chunk, if we cant we know the stream is over
+      m_stream->read(m_data.GetUnderlyingData(), DataBufferSize);
+      size_t bytesRead = static_cast<size_t>(m_stream->gcount());
 
-    // If we've read all of the underlying stream write the checksum trailing header
-    // the set that the chunked stream is over.
-    if (m_stream->eof() && !m_stream->bad() && (m_chunkingStream->eof() || m_chunkingStream->peek() == EOF)) {
-      return writeTrailer(dst, amountToRead);
-    }
+      if (bytesRead > 0) {
```

Comment:
it would be good to have less branching if possible. is this check really needed? bytesRead is a size_t cant be negative

Reply:
it can be zero however

Reply:
in theory it could be zero though i think

Reply:
yea, we can do without the check